tarofreighter/parcel.go

```diff
@@ -79,6 +79,10 @@ type AssetParcel struct {
 	// Dest is the address that should be used to satisfy the transfer.
 	Dest *address.Taro
 
+	// pkg is the actual package that is being shipped. This is not set in
+	// case we request delivery to a destination address.
+	pkg *sendPackage
```

Comment:
Perhaps use a distinct struct? Need to double check if `Dest` is still sent in this case, as otherwise might be some panics elsewhere since the field isn't set. 

We could also modify the actual channel use here to use a type param based on an interface that exposes the data needed via methods.

Reply:
Yeah, it wasn't really clear what fields would be accessed in what case. I refactored the parcels to make everything more generic and clear.